Added german translation

BlockListed opened the pull request:
Added a german translation.

adamviola commented:
Thanks for the translations! Do you think you can update them before we merge them in? I just made significant changes to the translation strings.

BlockListed commented:
I'm gonna update them now, thanks for reminding me!

BlockListed commented:
I updated the language resource bindings and added the additional translations. The commit also includes some information about how I decided to translate the fields.
